ProxyAgent/HttpClient/HttpClient.cs

```diff
@@ -220,11 +225,24 @@ public HttpClient(ILogger logger)
             }
         }
 
-        private static void SetServerSslSecurity(IHttpRequest request, HttpClientHandler clientHandler)
+        private void SetServerSslSecurity(IHttpRequest request, HttpClientHandler clientHandler)
         {
             if (request.Options.AllowInsecureSslServer && request.UsesSsl())
             {
-                clientHandler.ServerCertificateCustomValidationCallback = delegate { return true; };
+                clientHandler.ServerCertificateCustomValidationCallback += delegate(HttpRequestMessage sender, X509Certificate2 cert, X509Chain chain, SslPolicyErrors error)
+                {
+                    var sslThumbprint = cert.Thumbprint.ToLowerInvariant();
+                    var configThumbprint = this.config.Thumbprint.ToLowerInvariant();
+                    if (sslThumbprint  != configThumbprint)
+                    {
+                        this.log.Error("The thumbprint provided by the client don't match", () => new { sslThumbprint, configThumbprint });
```

Comment:
nit: change the msg to "The remote endpoint is using an unknown/invalid SSL certificate, the thumbprint of the certificate doesn't match the value in the configuration"